TestBootstrapDefaultKeyspaceGroup is unstable

## Flaky Test

### Which jobs are failing
```
--- FAIL: TestCommonTestSuite (18.04s)
    --- FAIL: TestCommonTestSuite/TestBootstrapDefaultKeyspaceGroup (2.23s)
        server_test.go:615: 
            	Error Trace:	/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:615
            	            				/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:627
            	Error:      	Should be empty, but was [{http://127.0.0.1:36161/ 0}]
            	Test:       	TestCommonTestSuite/TestBootstrapDefaultKeyspaceGroup
FAIL
```
### CI link
https://github.com/tikv/pd/actions/runs/9936180084/job/27444347281

### Reason for failure (if possible)


### Anything else

--- a/pkg/keyspace/tso_keyspace_group.go
+++ b/pkg/keyspace/tso_keyspace_group.go
@@ -211,6 +211,7 @@ func (m *GroupManager) allocNodesToAllKeyspaceGroups(ctx context.Context) {
 					log.Error("failed to alloc nodes for keyspace group", zap.Uint32("keyspace-group-id", group.ID), zap.Error(err))
 					continue
 				}
+				log.Info("alloc nodes for keyspace group", zap.Uint32("keyspace-group-id", group.ID), zap.Any("nodes", nodes))
 				group.Members = nodes
 			}
 		}

--- a/tests/integrations/mcs/tso/server_test.go
+++ b/tests/integrations/mcs/tso/server_test.go
@@ -598,22 +598,24 @@ func (suite *CommonTestSuite) TestAdvertiseAddr() {
 func (suite *CommonTestSuite) TestBootstrapDefaultKeyspaceGroup() {
 	re := suite.Require()
 
-	// check the default keyspace group
+	// check the default keyspace group and wait for alloc tso nodes for the default keyspace group
 	check := func() {
-		resp, err := tests.TestDialClient.Get(suite.pdLeader.GetServer().GetConfig().AdvertiseClientUrls + "/pd/api/v2/tso/keyspace-groups")
-		re.NoError(err)
-		defer resp.Body.Close()
-		re.Equal(http.StatusOK, resp.StatusCode)
-		respString, err := io.ReadAll(resp.Body)
-		re.NoError(err)
-		var kgs []*endpoint.KeyspaceGroup
-		re.NoError(json.Unmarshal(respString, &kgs))
-		re.Len(kgs, 1)
-		re.Equal(utils.DefaultKeyspaceGroupID, kgs[0].ID)
-		re.Equal(endpoint.Basic.String(), kgs[0].UserKind)
-		re.Empty(kgs[0].SplitState)
-		re.Empty(kgs[0].Members)
-		re.Empty(kgs[0].KeyspaceLookupTable)
+		testutil.Eventually(re, func() bool {
+			resp, err := tests.TestDialClient.Get(suite.pdLeader.GetServer().GetConfig().AdvertiseClientUrls + "/pd/api/v2/tso/keyspace-groups")
+			re.NoError(err)
+			defer resp.Body.Close()
+			re.Equal(http.StatusOK, resp.StatusCode)
+			respString, err := io.ReadAll(resp.Body)
+			re.NoError(err)
+			var kgs []*endpoint.KeyspaceGroup
+			re.NoError(json.Unmarshal(respString, &kgs))
+			re.Len(kgs, 1)
+			re.Equal(utils.DefaultKeyspaceGroupID, kgs[0].ID)
+			re.Equal(endpoint.Basic.String(), kgs[0].UserKind)
+			re.Empty(kgs[0].SplitState)
+			re.Empty(kgs[0].KeyspaceLookupTable)
+			return len(kgs[0].Members) == 1
+		})
 	}
 	check()
 

